qiskit/execute_function.py

```diff
@@ -275,6 +302,182 @@ def execute(
 
             job = execute(qc, backend, shots=4321)
     """
+    if basis_gates is not None:
+        warnings.warn(
+            "The basis_gates argument is deprecated as of Qiskit Terra 0.21.0, "
+            "and will be removed in a future release. If you require compiling to a "
+            "custom set of basis gates you should use the transpile() function instead.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+    if coupling_map is not None:
+        warnings.warn(
+            "The coupling_map argument is deprecated as of Qiskit Terra 0.21.0, "
+            "and will be removed in a future release. If you require compiling to a "
+            "custom coupling graph you should use the transpile() function instead.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+    if backend_properties is not None:
+        warnings.warn(
+            "The backend_properties argument is deprecated as of Qiskit Terra 0.21.0, "
+            "and will be removed in a future release. If you require compiling to a "
+            "custom BackendProperties object you should use the transpile() function instead.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+    if pass_manager is not None:
+        warnings.warn(
+            "The pass_manager argument is deprecated as of Qiskit Terra 0.21.0, "
+            "and will be removed in a future release. If you require compiling with a "
+            "custom PassManager object you should use the run() method on that custom "
+            "Passmanager directly instead of using execute.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+    if shots is not None:
+        if not hasattr(backend.options, "shots"):
+            warnings.warn(
+                "The shots argument is deprecated as of Qiskit Terra 0.21.0, "
+                "and will be removed in a future release. The backend you are running on does "
+                "not support setting the number of shots so this option will have no effect.",
+                DeprecationWarning,
+                stacklevel=2,
+            )
```

Comment:
If the `shots` keyword is deprecated, I think we need the warning in both paths? It'll just have a different message.  (Similar comment for `memory` and `seed_simulator` as well.)

On the other hand, `shots` may be one of the few options that could still be appropriate for `execute`.

*edit*: I just saw the extra comment in the release note about the first point here.  In this case, perhaps we might want to make the distinction in the docstring as well, since at the moment the documentation looks the same for a completely deprecated argument.

Reply:
Sure, I'll update the docstring some more. I honestly just got annoyed/overwhelmed by all the options here (you might notice the deprecation warnings get a bit less detailed the further down you go).

Reply:
Yeah, I totally understand that - there's hundreds of them.